Blackjack/Game/BlackjackGame.m

```diff
@@ -0,0 +1,246 @@
+    return 16;
+}
+
+- (NSArray *)currentActivePlayers {
+    NSMutableArray *activePlayers = [[NSMutableArray alloc] init];
+    for (Player *player in _players) {
+        if (player.isPlaying) {
+            [activePlayers addObject:player];
+        }
+    }
+    return activePlayers;
+}
+
+// MARK: - Init
+
+- (instancetype)initWith:(Deck *)deck {
+    self = [super init];
+    if (self) {
+        _deck = deck;
+    }
+    return self;
+}
+
+// MARK: - Game Methods
+
+- (void)startGame:(NSInteger)numberOfPlayers {
+    [self setupContestants:numberOfPlayers];
+    _gameState = collectBets;
+    [_delegate updateUI];
+}
+
+- (void)gameOver {
+//    [_dealer.cards removeAllObjects];
+}
+
+
+// MARK: - Game Logic
+
+-(void)setupContestants:(NSInteger) numberOfPlayers {
+    [_players removeAllObjects];
+    _players = [NSMutableArray arrayWithCapacity:numberOfPlayers];
+    
+    for (int i=0; i<numberOfPlayers; i++) {
+        NSString *name = [NSString stringWithFormat:@"Player #%d", i+1];
+        Player *player = [[Player alloc] initWith:name cards:[[NSMutableArray alloc] init] chips: 50 isPlaying:YES];
+        [_players addObject: player];
+    }
+    _currentPlayer = _players.firstObject;
+}
+
+- (void)setDecision:(enum Decision)decision {
+    switch (decision) {
+        case hit:
+            [_currentPlayer.cards addObject:[_deck drawRandomCard:YES]];
+            NSInteger cardsEvaluation = [self evaluateCardsFor:_currentPlayer];
+            [self handlePlayersLogicBasedOn: cardsEvaluation forPlayer: _currentPlayer];
+            
+            if (!_currentPlayer.isPlaying) {
+                [self nextPlayer];
+            }
+            break;
+            
+        case stand:
+            [self nextPlayer];
+            break;
+            
+        case surrender:
+            self.dealer.chips += _currentPlayer.betAmount;
+            _currentPlayer.betAmount = 0;
+            _currentPlayer.isPlaying = NO;
+            [self nextPlayer];
+            break;
+    }
+    [_delegate updateUI];
+}
+
+- (void)setBet:(NSInteger)amount {
+    if (_currentPlayer.chips >= amount) {
+        _currentPlayer.betAmount += amount;
+        _currentPlayer.chips -= amount;
+        [self nextPlayer];
+        [_delegate updateUI];
+    }
+}
+
+-(void)nextPlayer {
+    NSArray *currentActivePlayers = [self currentActivePlayers];
+
+    if (currentActivePlayers.count == 0) {
+        [self gameOver];
+        return;
+    }
+    NSInteger index = [_players indexOfObject:_currentPlayer];
+
+    if (index < _players.count - 1) {
+        _currentPlayer = _players[index + 1];
+    } else if (index == _players.count - 1) {
+        
+        [self progressGameState];
+    } else {
+        // TODO: handle case
+    }
+}
+
+// TODO : encapuslate some logic
+-(void)progressGameState {
+    switch (_gameState) {
+        case collectBets:
+            _gameState = dealCards;
+        
+        case dealCards:
```

Comment:
`case` with more than one line should be encapsulated with `{}`

See raywenderlich
[coding conventions](https://github.com/raywenderlich/objective-c-style-guide)

Reply:
Done